File: 94-binary-tree-inorder-traversal/94-binary-tree-inorder-traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var inorderTraversal = function(root) {
    let ans = []
    let curr = root
    
    while(curr){
        if(!curr.left){
            ans.push(curr.val)
            curr = curr.right
        }else{
            let temp = curr.left
            while(temp.right && temp.right!==curr){
                temp = temp.right
            }
            if(temp.right===null){
                temp.right=curr
                curr = curr.left
            }else{
                temp.right = null
                ans.push(curr.val)
                curr = curr.right
            }
        }
    }
    
    return ans
    
};

File: 94-binary-tree-inorder-traversal/94-binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        ans = []
        curr = root
        
        while curr is not None:
            if curr.left is None:
                ans.append(curr.val)
                curr = curr.right
            else:
                temp = curr.left
                
                while temp.right and temp.right!=curr:
                    temp = temp.right
                    
                if temp.right is None:
                    temp.right = curr
                    curr = curr.left
                else:
                    temp.right=None
                    ans.append(curr.val)
                    curr = curr.right
        
        return ans